kore/src/Kore/Step/Strategy.hs

```diff
@@ -368,6 +376,11 @@ emptyExecutionGraph config =
 -}
 data GraphSearchOrder = BreadthFirst | DepthFirst deriving Eq
 
+data LimitExceeded = LimitExceeded | Other
```

Comment:
What is `Other` for? It would also be nice if the exception type captured the execution queue when it is thrown.

Reply:
I know I did an ugly thing, but I did this in order to distinguish between the failing of the function and the failing of the test caused by the success of the function. I mostly pushed this to save my work. It was the only way I was able to make it work in almost 6 hours. I'll try to come with a better solution

Reply:
I think I already have a better idea :))

Reply:
It looks like you figured this out below, but you can `catch` or `handle` the `LimitExceeded` exception and use `assertFailure` if the exception is not thrown.